Add kafka-connector-configuration.md (#1142)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/kafka-connector/kafka-connector-configuration.md

# Kafka Connector Configuration

[[Prerequisites]](#Prerequisites) [[Initializing the Connector]](#initializing-the-connector)

The Kafka connector allows you to access the Kafka Producer API through WSO2 EI and acts as a message producer that facilitates message publishing. The Kafka connector sends messages to the Kafka brokers.

> Kafka is a distributed publish-subscribe messaging system that maintains feeds of messages in topics. Producers write data to topics and consumers read from topics. For more information on Apache Kafka, see [Apache Kafka documentation](http://kafka.apache.org/documentation.html).

Reviewer: Kafka is a distributed publish-subscribe messaging system that maintains the feeds of

Author: Updated the content of the first paragraph.

Kafka is a distributed, partitioned, replicated commit log service. It provides the functionality of a publish-subscribe messaging system with a unique design. Kafka mainly operates based on a topic model. A topic is a category or feed name to which records get published. Topics in Kafka are always multi-subscriber.

> 

Reviewer 2: Shall we remove these unnecessary empty lines. Shall we update the entire 2 documents?

Author: Done.

>The recommended version is [kafka](https://www.apache.org/dyn/closer.cgi?path=/kafka/1.0.0/kafka_2.12-1.0.0.tgz). For all available versions of Kafka that you can download, see https://kafka.apache.org/downloads. The recommended Java version is 1.8.

To configure the Kafka connector, copy the following client libraries from the `<KAFKA_HOME>/lib` directory to the `<PRODUCT_HOME>/repository/components/lib` directory.

* [kafka_2.12-1.0.0.jar](https://mvnrepository.com/artifact/org.apache.kafka/kafka_2.12/1.0.0)  
* [kafka-clients-1.0.0.jar](https://mvnrepository.com/artifact/org.apache.kafka/kafka-clients/1.0.0)
* [metrics-core-2.2.0.jar](https://mvnrepository.com/artifact/com.yammer.metrics/metrics-core/2.2.0)
* [scala-library-2.12.3.jar](https://mvnrepository.com/artifact/org.scala-lang/scala-library/2.12.3)
* [zkclient-0.10.jar](https://mvnrepository.com/artifact/com.101tec/zkclient/0.10)
* [zookeeper-3.4.10.jar](https://mvnrepository.com/artifact/org.apache.zookeeper/zookeeper/3.4.10)

---
__Note :__

> When configuring the Kafka Inbound Operation, navigate to https://store.wso2.com/store/assets/esbconnector/details/kafka, and click **Download Inbound Endpoint** to download the inbound JAR file and add the downloaded .jar file to the <PRODUCT_HOME>/dropins directory.

Reviewer 3: Let's use the following format:

!!! Note
    When configuring the Kafka Inbound Operation, navigate to https://store.wso2.com/store/assets/esbconnector/details/kafka, and click **Download Inbound Endpoint** to download the inbound JAR file and add the downloaded .jar file to the <PRODUCT_HOME>/dropins directory.

Author: Done.

---

Run the following command to start the ZooKeeper server:
```
bin/zookeeper-server-start.sh config/zookeeper.properties
```

Run the following command to start the Kafka server:
```
bin/kafka-server-start.sh config/server.properties
```

> Now that you have connected to Kafka, you can start publishing and consuming messages using the Kafka brokers. For more information, see 

Reviewer 3: Link is missing? Do you plan to add this later?

Author: Yes.Will add this links later.

# Kafka Connector Configuration

> [[Setting up the environment]](#Setting-up-the-environment) 

Reviewer 3: Let's remove this line.

Author: Removed.

The Kafka connector allows you to access the Kafka Producer API through WSO2 EI and acts as a message producer that facilitates message publishing. The Kafka connector sends messages to the Kafka brokers.

> The Kafka is a distributed publish-subscribe messaging system that maintains feeds of messages in topics. Producers write data to topics and consumers read from topics. For more information on Apache Kafka, see [Apache Kafka documentation](http://kafka.apache.org/documentation.html).

Reviewer 3: ```suggestion 
Kafka is a distributed publish-subscribe messaging system that maintains feeds of messages in topics. Producers write data to topics and consumers read from topics. For more information on Apache Kafka, see [Apache Kafka documentation](http://kafka.apache.org/documentation.html).
```

Author: Done.

> It provides the functionality of a publish-subscribe messaging system with a unique design.The Kafka mainly operates based on a topic model. A topic is a category or feed name to which records get published. Topics in Kafka are always multi-subscriber.

Reviewer 3: ```suggestion
Kafka mainly operates based on a topic model. A topic is a category or feed name to which records get published. Topics in Kafka are always multi-subscriber.
```

Removing this first line since it is repetitive.

Author: Done.

## Setting up the environment

> To use the Kafka connector, download and install [Apache Kafka](http://kafka.apache.org/downloads.html).

Reviewer 3: Lets add something that says you also need WSO2 EI and we refer to that location as <PRODUCT_HOME>.

Author: Added new line for this.

>The recommended version is [kafka](https://www.apache.org/dyn/closer.cgi?path=/kafka/1.0.0/kafka_2.12-1.0.0.tgz). For all available versions of Kafka that you can download, see https://kafka.apache.org/downloads. The recommended Java version is 1.8.

> To configure the Kafka connector, copy the following client libraries from the <KAFKA_HOME>/lib directory to the <ESB_HOME>/repository/components/lib directory.

Reviewer 3: ```suggestion
To configure the Kafka connector, copy the following client libraries from the `<KAFKA_HOME>/lib` directory to the `<PRODUCT_HOME>/repository/components/lib` directory.
```

Author: Done.

* [kafka_2.12-1.0.0.jar](https://mvnrepository.com/artifact/org.apache.kafka/kafka_2.12/1.0.0)  
* [kafka-clients-1.0.0.jar](https://mvnrepository.com/artifact/org.apache.kafka/kafka-clients/1.0.0)
* [metrics-core-2.2.0.jar](https://mvnrepository.com/artifact/com.yammer.metrics/metrics-core/2.2.0)
* [scala-library-2.12.3.jar](https://mvnrepository.com/artifact/org.scala-lang/scala-library/2.12.3)
* [zkclient-0.10.jar](https://mvnrepository.com/artifact/com.101tec/zkclient/0.10)
* [zookeeper-3.4.10.jar](https://mvnrepository.com/artifact/org.apache.zookeeper/zookeeper/3.4.10)

---
__Note :__

> When you Configuring the Kafka Inbound Operation, navigate to https://store.wso2.com/store/assets/esbconnector/details/kafka, and click lick Download Inbound Endpoint to download the inbound JAR file and add the downloaded .jar file to the <EI_HOME>/dropins directory.

Reviewer 3: ```suggestion
When configuring the Kafka Inbound Operation, navigate to https://store.wso2.com/store/assets/esbconnector/details/kafka, and click **Download Inbound Endpoint** to download the inbound JAR file and add the downloaded .jar file to the <PRODUCT_HOME>/dropins directory.
```

Author: Done.

---

Run the following command to start the ZooKeeper server:
```
bin/zookeeper-server-start.sh config/zookeeper.properties
```

Run the following command to start the Kafka server:
```
bin/kafka-server-start.sh config/server.properties
```
Now that you have connected to Kafka, you can start publishing and consuming messages using the Kafka brokers. For more information, see 

> [Publishing Messages using Kafka](kafka-connector-producer-example.md).

Reviewer 3: We have not added these yet?

Author: Removed links.
